projects/media-viewer/src/lib/annotations/comment-set/comment-set-header/comment-set-header.component.ts

```diff
@@ -1,27 +1,41 @@
-import { Component, EventEmitter, Input, OnChanges, Output, SimpleChanges, ViewEncapsulation } from '@angular/core';
-import { AnnotationSet } from '../../annotation-set/annotation-set.model';
+import {Component, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges, ViewEncapsulation} from '@angular/core';
 import { Annotation } from '../../annotation-set/annotation-view/annotation.model';
+import {select, Store} from '@ngrx/store';
+import * as fromStore from '../../../store/reducers';
+import * as fromTagSelectors from '../../../store/selectors/tags.selectors';
+import * as fromAnnoSelector from '../../../store/selectors/annotations.selectors';
 
 @Component({
   selector: 'mv-comment-set-header',
   templateUrl: './comment-set-header.component.html',
   encapsulation: ViewEncapsulation.None
 })
-export class CommentSetHeaderComponent implements OnChanges {
+export class CommentSetHeaderComponent implements OnInit {
 
-  @Input() public readonly annotationSet: AnnotationSet;
   @Input() public showCommentSummary: boolean;
   @Output() public readonly showCommentSummaryDialog = new EventEmitter();
 
-  tabs = ['comments', 'search'];
+  tabs = [{label: 'comments'}, {label: 'filter'}, {label: 'search'}];
   tabSelected: string;
+  isFiltered: boolean;
   navigationList: Annotation[];
 
-  ngOnChanges(changes: SimpleChanges): void {
-    if (changes.annotationSet && this.annotationSet) {
-      this.navigationList = [...this.annotationSet.annotations]
-        .filter(annotation => annotation.comments && annotation.comments.length > 0);
-    }
+  constructor(private store: Store<fromStore.State>) {}
+
+
+  ngOnInit(): void {
+    this.store.pipe(select(fromTagSelectors.getTagFilters)).subscribe(formData => {
+     this.isFiltered = !formData.length;
+     this.tabs = [...this.tabs].map((tab) => {
+       return !this.isFiltered && tab.label === 'filter' ? {...tab, isFiltered: true} : {...tab, isFiltered: false};
+     });
+    });
+
+    this.store.pipe(select(fromAnnoSelector.getFilteredAnnotations)).subscribe(filteredAnno => {
+      this.navigationList = filteredAnno.length ?
+        filteredAnno.filter(annotation => annotation.comments && annotation.comments.length > 0) : [];
+
```

Comment:
is it not worth having a selector for this?

Reply:
yeap - good spot